source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -901,6 +988,41 @@ again, so ensuring the socket is torn down does not need to happen
 immediately and can happen at a later time, either via async I/O or a
 background thread. 
 
+Why can the pool be paused?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The distinction between the "paused" state and the "ready" state allows the pool
+to determine whether or not the endpoint it is associated with is available or
+not. This enables the following behaviors:
+
+1. The pool can halt the creation of background connection establishments until
+   the endpoint becomes available again. Without the "paused" state, the pool
+   would have no way of determining when to begin establishing background
+   connections again, so it would just continually attempt, and often fail, to
+   create connections until minPoolSize was satisfied, even after repeated
+   failures. This could unnecessarily waste resources both server and driver side.
+
+2. The pool can evict requests that enter the WaitQueue after the pool was
+   cleared but before the server was in a known state again. Such requests can
+   occur when a server is selected at the same time as it becomes marked as
+   Unknown in highly concurrent workloads. Without the "paused" state, the pool
+   would attempt to service these requests, since it would assume they were
+   routed to the pool because its endpoint was available, not because of a race
+   between SDAM and Server Selection. These requests would then likely fail with
+   potentially high latency, again wasting resources both server and driver side.
+
+Why not emit PoolCleared events when clearing a paused pool?
```

Comment:
This change breaks the following SDAM tests:
- https://github.com/mongodb/specifications/blob/f123e3a/source/server-discovery-and-monitoring/tests/integration/isMaster-command-error.yml#L39
- https://github.com/mongodb/specifications/blob/f123e3a/source/server-discovery-and-monitoring/tests/integration/isMaster-network-error.yml#L38
- https://github.com/mongodb/specifications/blob/f123e3a/source/server-discovery-and-monitoring/tests/integration/isMaster-timeout.yml#L39

See the follow test failure: https://evergreen.mongodb.com/task/mongo_python_driver_tests_python_version_rhel62_test_ssl__platform~rhel62_auth~auth_ssl~ssl_python_version~2.7_coverage~coverage_test_latest_standalone_patch_61232b7f21d8f0ad9023c8ef0912017d062fafc0_5fd02078c9ec4456b5c14943_20_12_09_00_55_21

Since the pool initially starts in the paused state the PoolClearedEvent which those tests expect is no longer emitted. I propose we revert this change and continue to emit PoolCleared events when clearing a paused pool. This way we won't create a dependency on https://jira.mongodb.org/browse/DRIVERS-1314.

Reply:
I'd rather update the tests than update the proposed driver functionality. After reading over the various tickets there, it seems Go's not having SDAM monitoring events was one of the hangups, but that doesn't seem to be an issue any more. The other hangup is that we wanted to resolve SPEC-1474 first, but now that we have the "paused" state, we can resolve DRIVERS-1314 without actually addressing SPEC-1474, since regardless of whether the pool is created or not, no PoolClearedEvent will be emitted anyways. The test update should then be pretty simple and localized to just removing these PoolClearedEvents.

Reply:
Okay SGTM, let's remove the PoolClearedEvents from the failing "command/network error/timeout on Monitor handshake" tests.

Reply:
Done